src/Microsoft.IdentityModel.Tokens/Exceptions/SecurityTokenInvalidCloudInstanceNameException.cs

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.Serialization;
+
+namespace Microsoft.IdentityModel.Tokens
+{
+    /// <summary>
+    /// This exception is thrown when the cloud instance name of the signing key was not matched with the cloud instance name from configuration.
+    /// </summary>
+    [Serializable]
+    public class SecurityTokenInvalidCloudInstanceNameException : SecurityTokenInvalidSigningKeyException
+    {
```

Comment:
Suggest the name: SecurityTokenInvalidCloudInstanceException  (drop the 'Name') as there may be other reasons in the future.

@GeoK not sure about the need to inherit from SecurityTokenInvalidSigningKeyException, perhaps SecurityTokenException

Reply:
Renamed class.
in my opinion SecurityTokenInvalidCloudInstanceNameException is still SecurityTokenInvalidSigningKeyException but more specific

Reply:
Discussed this with Brent offline - it is ok as-is.